Add a first draft of the README

peteryates opened the pull request:
First crack at a semi-useful README. Feedback and additions welcome.

cc @rjlynch @kylewelsby @robhettrick 

Fixes #20

peteryates committed:
Add a first draft of the README

peteryates committed:
Add a schema diagram

peteryates committed:
Remove unnecessary word

kylewelsby commented on README.md:
```diff
@@ -2,69 +2,110 @@
+ * Ruby 2.6.5
+ * PostgreSQL (with an account that has superuser privileges)
+ * NodeJS
+ * NPM
```
```suggestion
```

peteryates replied:
:+1:

kylewelsby commented on README.md:
```diff
@@ -2,69 +2,110 @@
+
+ * Ruby 2.6.5
+ * PostgreSQL (with an account that has superuser privileges)
+ * NodeJS
```
```suggestion
 * NodeJS ^12.14.1
```

kylewelsby commented on README.md:
```diff
@@ -2,69 +2,110 @@
-bundle exec rspec
+
+If your database user needs a password, set that first by editing
+`config/database.yml`.  Now run the app:
```
```suggestion
_If your database user needs a password, set that first by editing
`config/database.yml`._  

Now run the app:
```

kylewelsby commented on README.md:
```diff
@@ -2,69 +2,110 @@
-- Dotenv (managing environment variables)
-- Travis with Heroku deployment
+The application is a very standard [Rails](https://www.rubyonrails.org/) application and no
+special steps should be required in order to get up and running.
```
```suggestion
The application is a standard [Ruby on Rails](https://www.rubyonrails.org/) application requiring no additional steps to setup.
```

kylewelsby commented on README.md:
```diff
@@ -2,69 +2,110 @@
-3. Run `npm install` to vendor node dependencies
-4. Run `bundle exec rails webpacker:compile` to compile assets
-5. Run `cf push` to push the app to Cloud Foundry Application Runtime
+## Lingo
```
```suggestion
## Glossary
```

Consider moving this and contributing to `.github/CONTRIBUTING.md` 

examples:-

- https://github.com/DFE-Digital/presentation-merger/blob/master/.github/CONTRIBUTING.md
- https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md
- https://github.com/facebook/jest/blob/master/CONTRIBUTING.md

peteryates replied:
Yep, good point, will do

kylewelsby commented on README.md:
````diff
@@ -2,69 +2,110 @@
+
+```bash
+$ bundle exec rake
+```
````
Include license

```suggestion
## 🎓 License

Unless stated otherwise, the codebase is released under the MIT License. This covers both the codebase and any sample code in the documentation. The documentation is © Crown copyright and available under the terms of the Open Government 3.0 licence.
```

peteryates replied:
Added MIT for now, we won't need OGL [unless we start publishing documentation externally](https://gds-way.cloudapps.digital/manuals/licensing.html#copyright-notice)

kylewelsby reviewed (commented):
I have a few suggestions, mostly about the voice.

Also would be ideal to include better documentation about contributing and license.

peteryates committed:
Make text tweaks based on @kylewelsby's feedback

peteryates committed:
Move contribution guide to its own file

This is based heavily on @kylewelsby's
https://github.com/DFE-Digital/presentation-merger/blob/master/.github/CONTRIBUTING.md

peteryates committed:
Add MIT Licence

kylewelsby commented on LICENCE:
```diff
@@ -0,0 +1,19 @@
+Copyright 2020 Department for Education Digital
```
After checking a bunch of repositories on https://github.com/DFE-Digital I found this MIT copyright is used most. 
```suggestion
MIT License

Copyright (c) 2020 Crown Copyright (Department for Education)
```

peteryates replied:
Ok, fixed this. I copied the text directly from the [OSI](https://opensource.org/licenses/MIT).

kylewelsby commented on README.md:
```diff
@@ -2,69 +2,103 @@
+
+Ensure you have the following installed on your machine:
+
+ * Ruby 2.6.5
```
```suggestion
 * Ruby `2.6.5`
```

kylewelsby reviewed (approved):
These are a few suggestions but overall I'm OK with this. 

We can improve over-time

peteryates committed:
Remove LICENSE file and update copyright info

peteryates committed:
Format Ruby version number
